### PR TITLE
feat(fractional_differencing): create frac diff plugin

### PR DESCRIPTION
**Pull request by ngriffiths13**
The transformer implementation of frac diff was also replaced with the plugin implementation. The api did not change however.

## What does this implement/fix? Explain your changes.
Fractional differencing plugin

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@ functime_embs
 catboost_info
 bin/
 .data
+pdm.lock
```
Comment (baggiponte) [inline .gitignore:6]: Uh you use PDM? that's great! I would add this in another place though.

**Review by baggiponte — CHANGES_REQUESTED** [review]
Nothing to add about the implementation. Kudos for writing the docstring as well. Could it be possible to add some tests too?

**Comment by abstractqqq** [comment]
I agree. Let's add one test and have this approved.

**Comment by ngriffiths13** [comment]
> Nothing to add about the implementation. Kudos for writing the docstring as well. Could it be possible to add some tests too?

So there were already tests here from the previous implementation. They still pass. Do you want me to write an explicit test for the plugin? The transformer that uses the plugin is tested

**Comment by baggiponte** [comment]
> I don't think this is related to this PR. But does anybody know what is aeon?
 
It's a fork of functime.

**Comment by baggiponte** [comment]
LGTM :)

**Pull request merged**